test/gurka/test_traffic.cc

```diff
@@ -0,0 +1,179 @@
+#include "gurka.h"
+#include "microtar.h"
+#include <gtest/gtest.h>
+
+#include <valhalla/baldr/graphreader.h>
+#include <valhalla/baldr/traffictile.h>
+
+#include <boost/property_tree/ptree.hpp>
+
+#include <cmath>
+#include <sstream>
+#include <sys/mman.h>
+#include <sys/stat.h>
+
+using namespace valhalla;
+
+/** Copy of raw header for use with sizeof() **/
+typedef struct {
+  char name[100];
+  char mode[8];
+  char owner[8];
+  char group[8];
+  char size[12];
+  char mtime[12];
+  char checksum[8];
+  char type;
+  char linkname[100];
+  char _padding[255];
+} mtar_raw_header_t_;
+
+/*************************************************************/
+TEST(Traffic, BasicUpdates) {
+
+  const std::string ascii_map = R"(
+    A----B----C
+         |    |
+         D----E)";
+
+  const gurka::ways ways = {{"AB", {{"highway", "primary"}, {"maxspeed", "10"}}},
+                            {"BC", {{"highway", "primary"}, {"maxspeed", "10"}}},
+                            {"BD", {{"highway", "primary"}, {"maxspeed", "10"}}},
+                            {"CE", {{"highway", "primary"}, {"maxspeed", "10"}}},
+                            {"DE", {{"highway", "primary"}, {"maxspeed", "10"}}}};
+
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, 100);
+  std::string tile_dir = "test/data/traffic_basicupdates";
+  auto map = gurka::buildtiles(layout, ways, {}, {}, tile_dir);
+
+  std::string traffic_extract = tile_dir + "/traffic.tar";
+
+  {
+    mtar_t tar;
+    auto tar_open_result = mtar_open(&tar, traffic_extract.c_str(), "w");
+    ASSERT_EQ(tar_open_result, MTAR_ESUCCESS);
+
+    baldr::GraphReader reader(map.config.get_child("mjolnir"));
+
+    auto tile_ids = reader.GetTileSet();
```

Comment:
Could you add a comment explaining the high  level approach done below? I.e. filling in the live data file which is usually done by the sidecar?